### PR TITLE
Refactor Android module

### PR DESCRIPTION
**Pull request by DmitryTop**
Improved native Android module to be able to use Twiliio API outside of React components

### CODE REVIEW

**Comment by DmitryTop** [comment]
Breaking changes:

* Existing `TwilioVideo` was renamed to `TwilioVideoView`
* Added `TwilioVideo` class to use native api outside of Views

API:

```typescript
  interface TwilioVideo {
    connect: (
      roomName: string,
      accessToken: string,
      options: iOSConnectParams | androidConnectParams,
    ) => void;

    disconnect: () => void;

    flipCamera: () => void;

    getStats: () => void;

    sendString: (message: string) => void;

    setLocalVideoEnabled: (enabled: boolean) => Promise<boolean>;

    setLocalAudioEnabled: (enabled: boolean) => Promise<boolean>;

    onRoomDidConnect: RoomEventCb;

    onRoomDidDisconnect: RoomErrorEventCb;

    onRoomDidFailToConnect: RoomErrorEventCb;

    onRoomParticipantDidConnect: ParticipantEventCb;

    onRoomParticipantDidDisconnect: ParticipantEventCb;

    onParticipantAddedVideoTrack: TrackEventCb;

    onParticipantAddedDataTrack: TrackEventCb;

    onParticipantRemovedDataTrack: TrackEventCb;

    onParticipantRemovedVideoTrack: TrackEventCb;

    onParticipantAddedAudioTrack: TrackEventCb;

    onParticipantRemovedAudioTrack: TrackEventCb;

    onParticipantRemovedDataTrack: TrackEventCb;

    onDataTrackMessageReceived: MessageReceivedEventCb;

    onStatsReceived: StatsReceivedEventCb;
  }

```

File: index.d.ts [hunk 1]
```diff
@@ -85,23 +133,21 @@ declare module "react-native-twilio-video-webrtc"{
...
-      // if audioBitrate OR videoBitrate is provided, you must provide both
-      audioBitrate?: number;
-      videoBitrate?: number;
-    }
-  }
+    enableH264Codec?: boolean;
+    // if audioBitrate OR videoBitrate is provided, you must provide both
+    audioBitrate?: number;
+    videoBitrate?: number;
+  };
```
Comment (andrii-kosovych) [inline index.d.ts:140]: I'm wondering why did you extract fields outside of `encodingParameters`?

**Comment by andrii-kosovych** [comment]
👏  looks beautiful to me!

File: index.d.ts (hunk 1 above)
Comment (a-morenovalls) [reply index.d.ts:140]: I was wondering the same...

**Comment by a-morenovalls** [comment]
LGTM! This will be a good improvement!

File: index.d.ts (hunk 1 above)
Comment (DmitryTop) [reply index.d.ts:140]: I don't see a reason for this nesting. All connection options can be combined in one map.

**Pull request merged**